### PR TITLE
PLANET-3613 Show author ideas on author profile page

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**
[JIRA 3613](https://jira.greenpeace.org/browse/PLANET-3613)

image

### CODE REVIEW

File: templates/author.twig [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+					<div class="col-lg-8 col-md-6 col-sm-12 mt-md-0">
+						<p class="d-md-block author-bio">{{ fn('wpautop', fn('the_author_meta', 'description', author.id))|e('wp_kses_post')|raw }}</p>
+					</div>
+				</div>
+			{% endif %}
+		</div>
+	</header>
+
+	<div class="page-template">
+		<h3>{{ __( 'Posts by ', 'planet4-child-theme-handbook' ) }} {{ author.name }}</h3>
```
Comment (angtheod) [inline templates/author.twig]: Could we add here `{% if posts|length > 0 %}` the way we check for ideas?

**Comment by angtheod** [comment]
In the screenshot the Posts thumbnails do not get rendered. Is this because you did not copy the images or is there an issue?

File: templates/tease-idea.twig [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<li id="result-row-{{ idea.ID }}" class="media search-result-list-item">
```
Comment (angtheod) [inline templates/tease-idea.twig]: Is this all the full info we want for Ideas? Do we want also perhaps to show the Status of the Idea (e.g. Approved, Completed, Declined, Open, ...)?

**Comment by angtheod** [comment]
Did you test if pagination works correctly? Because I see pagination only for Posts not for Ideas, so maybe we need to check that it works nice.

File: author.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	$post_args['author'] = $wp_query->query_vars['author'];
+	$idea_args['author'] = $wp_query->query_vars['author'];
+}
+
+if ( get_query_var( 'page' ) ) {
+	$templates          = [ 'tease-author.twig' ];
+	$page               = get_query_var( 'page' );
+	$post_args['paged'] = $page;
+
+	$posts = new PostQuery( $post_args, 'P4_Post' );
```
Comment (angtheod) [inline author.php]: This should work as well `$posts = new P4_Post( $post_args )`. I think it is more comprehensible/readable this way.

**Comment by koyan** [comment]
@angtheod We will not implement pagination. We are going for a very simple solution at this stage.

**Comment by koyan** [comment]
FYI. These are the requirements that we asked
Screenshot 2019-07-01 at 11 37 38

**Comment by sagarsdeshmukh** [comment]
> In the screenshot the Posts thumbnails do not get rendered. Is this because you did not copy the images or is there an issue?

Hi Angelos, Yes I copied the handbook docker image and DB on my local docker env but not the images. No issues with thumbnail images.

File: author.php (hunk 3 above)
Comment (sagarsdeshmukh) [reply author.php]: @angtheod - The `P4_Post()` class expects the Post id as argument, here we need author specific 10 posts per page. do you think we could use `$posts = new P4_Post( $post_args )` for same? Please clarify if I misunderstood your point :)

**Pull request merged**